file-system.ts

```diff
@@ -443,5 +444,28 @@ export class FileSystem implements IFileSystem {
 		}
 		return foundFiles;
 	}
+
+	public getFileShasum(fileName: string, encoding?: string): IFuture<string> {
+		let future = new Future<string>();
+		encoding = encoding || "sha1";
+		let logger: ILogger = this.$injector.resolve("logger");
+		let shasumData = crypto.createHash(encoding);
+		let fileStream = this.createReadStream(fileName);
+		fileStream.on('data',(data: any) => {
```

Comment:
https://nodejs.org/api/stream.html#stream_event_data
according to this it could be buffer or string, so I prefer to keep it any